pkg/apis/resource/validation/validation.go

```diff
@@ -266,6 +269,21 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		func(consumer resource.ResourceClaimConsumerReference) (types.UID, string) { return consumer.UID, "uid" },
 		fldPath.Child("reservedFor"))...)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
```

Comment:
@thockin @liggitt IIRC we don't need to feature gate here in validation, the fields will be dropped in strategy when the feature is off, can you please confirm?

Reply:
correct, we generally *don't* want to conditionally validate based on feature gates, we want to validate based on presence of data in the fields

Reply:
It's slightly more subtle than Jordan makes it sound.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md#using-a-feature-gate-in-code

Reply:
Thank you for the explanation, fixed. I removed the feature-gates in the validation (I still kept the feature-gate tests)